src/lib/input/input.ts

```diff
@@ -41,334 +36,104 @@ const MD_INPUT_INVALID_INPUT_TYPE = [
  */
 @Component({
   moduleId: module.id,
-  selector: 'md-input, md-textarea',
+  selector: 'input[md-input], textarea[md-textarea]',
   templateUrl: 'input.html',
   styleUrls: ['input.css'],
-  providers: [MD_INPUT_CONTROL_VALUE_ACCESSOR],
-  host: {'(click)' : 'focus()'},
   encapsulation: ViewEncapsulation.None,
+  providers: [
+    { provide: MD_PLACEHOLDER_HOST_TOKEN, useExisting: forwardRef(() => MdInput) }
+  ],
+  host: {
+    // This is to remove the properties of the `input md-input` itself. We still want to use them
+    // as an @Input though, so we use HostBinding.
+    'class': '',
+    'style': '',
+    'attr.align': ''
+  }
 })
-export class MdInput implements ControlValueAccessor, AfterContentInit, OnChanges {
-  private _focused: boolean = false;
-  private _value: any = '';
-
-  /** Callback registered via registerOnTouched (ControlValueAccessor) */
-  private _onTouchedCallback: () => void = noop;
-  /** Callback registered via registerOnChange (ControlValueAccessor) */
-  private _onChangeCallback: (_: any) => void = noop;
-
-  /**
-   * Aria related inputs.
-   */
-  @Input('aria-label') ariaLabel: string;
-  @Input('aria-labelledby') ariaLabelledBy: string;
-
-  private _ariaDisabled: boolean;
-  private _ariaRequired: boolean;
-  private _ariaInvalid: boolean;
-
-  @Input('aria-disabled')
-  get ariaDisabled(): boolean { return this._ariaDisabled; }
-  set ariaDisabled(value) { this._ariaDisabled = coerceBooleanProperty(value); }
-
-  @Input('aria-required')
-  get ariaRequired(): boolean { return this._ariaRequired; }
-  set ariaRequired(value) { this._ariaRequired = coerceBooleanProperty(value); }
+export class MdInput implements OnInit, MdPlaceholderHost {
+  @ViewChild(DomProjectionHost) _host: DomProjectionHost;
+  @ViewChild('suffix') _suffix: TemplateRef<any>;
+  @ViewChild('prefix') _prefix: TemplateRef<any>;
 
-  @Input('aria-invalid')
-  get ariaInvalid(): boolean { return this._ariaInvalid; }
-  set ariaInvalid(value) { this._ariaInvalid = coerceBooleanProperty(value); }
-
-  /**
-   * Content directives.
-   */
-  @ContentChild(MdPlaceholder) _placeholderChild: MdPlaceholder;
-  @ContentChildren(MdHint) _hintChildren: QueryList<MdHint>;
+  @ViewChild('suffixWrapper', { read: PortalHostDirective }) placeholderPortalHost: PortalHost;
+  private _focused: boolean = false;
 
-  /** Readonly properties. */
-  get focused() { return this._focused; }
-  get empty() { return (this._value == null || this._value === '') && this.type !== 'date'; }
-  get characterCount(): number {
-    return this.empty ? 0 : ('' + this._value).length;
+  @Input('class') _cssClass: string = '';
+  @Input('style') _cssStyle: string = '';
+  get _safeCssStyle(): SafeStyle {
+    return this._dom.bypassSecurityTrustStyle(this._cssStyle || '');
   }
-  get inputId(): string { return `${this.id}-input`; }
+  @HostBinding('attr.class') get _attrClass(): any { return null; }
+  @HostBinding('attr.style') get _attrStyle(): any { return null; }
```

Comment:
I also just thought about the fact that `ngClass`, `ngStyle`, `[class.whatever]`, and `[style.whatever]` will all potentially be a problem here. We should chat about dealing with these (potentially with Tobias)